content/sensu-go/5.17/guides/secrets-management-vault.md

```diff
@@ -0,0 +1,212 @@
+      "max_retries": 2,
+      "timeout": "20s",
+      "rate_limiter": {
+        "limit": 10.0,
+        "burst": 100
+      }
+    }
+  }
+}' \
+http://127.0.0.1:8080/api/enterprise/secrets/v1/providers/vault
+
+HTTP/1.1 200 OK
+{{< /highlight >}}
+
+## Create your secrets
+
+To create your secrets provider `vault`, send a PUT request to the [`/secrets/:secret` API endpoint][14].
+
+Secrets that target a HashiCorp Vault must start with the word `secret`.
+In this example, the name of the secret is `ansible`.
+The `ansible` secret contains a value called `token`, which is the password to our database.
+
+{{< highlight shell >}}
+curl -X PUT \
+-H "Authorization: Bearer $SENSU_ACCESS_TOKEN" \
+-H 'Content-Type: application/json' \
+-d '{
+  "type": "Secret",
+  "api_version": "secrets/v1",
+  "metadata": {
+    "name": "VAULT_TOKEN",
+    "namespace": "default"
+  },
+  "spec": {
+    "id": "secret/ansible#token",
+    "provider": "vault"
+  }
+}' \
+http://127.0.0.1:8080/api/enterprise/secrets/v1/namespaces/default/secrets/VAULT_TOKEN
+
+HTTP/1.1 200 OK
+{{< /highlight >}}
+
+## Add a check, handler, and mutator
+
+Here's an overview of how the check, handler, and mutator will work:
+
+- The check will **ADD**.
+- The handler will **ADD**.
+- The mutator will **ADD**.
+- All three will require your backend to request secrets from your HashiCorp Vault secrets provider.
+
+**Unsure what might be a realistic and useful example to use here. How should the check, handler, and mutator work together?**
+
+Secrets are exposed to Sensu services at runtime as environment variables.
+The `secrets` scope of the request payload and the environment variables are automatically redacted from all Sensu service logs and dashboards to prevent secret leakage.
+
+**Is the above true for Vault secrets?**
+
+The Sensu backend will cache fetched secrets in memory, with no persistence to a Sensu datastore or file on disk.
+Sensu deletes secrets provided via a lease with a [lease duration][12] from Sensu's in-memory cache after the configured number of seconds, prompting the Sensu backend to request the secret again.
+
+**I borrowed the examples from the check, handler, and mutator references as placeholders below.**
+
+First, to create the check, run:
+
+{{< highlight shell >}}
+cat << EOF | sensuctl create
+---
+type: CheckConfig
+api_version: core/v2
+metadata:
+  name: ansible-tower
+  namespace: ops
+spec:
+  check_hooks: null
+  command: sensu-ansible.sh $VAULT_TOKEN
+  secrets:
+  - name: VAULT_TOKEN
+    secret: sensu-vault-token
+EOF
+{{< /highlight >}}
+
+Second, create the handler definition. Run:
+
+{{< highlight shell >}}
+cat << EOF | sensuctl create
+---
+type: Handler 
+api_version: core/v2 
+metadata:
+  name: ansible-tower
+  namespace: ops
+spec: 
+  type: pipe
+  command: sensu-ansible-handler -h $VAULT_HOST -t $VAULT_TOKEN
+  secrets:
+  - name: VAULT_HOST
+    secret: sensu-vault-host
+  - name: VAULT_TOKEN
+    secret: sensu-vault-token
+EOF
```

Comment:
Hmm these secrets don't seem to exist

Reply:
Yes -- I just made up placeholders because I wasn't sure what to put here in the check, handler, and mutator definitions.

Maybe this could be part of the minimum viable Vault setup (maybe at the beginning of this guide)?